### DISCUSSION TITLE
en/latest/getting-started/installation/

### DISCUSSION THREAD

**Discussion by giscus** [EndstoneMC/endstone#9] [Giscus]
# en/latest/getting-started/installation/

https://endstone.readthedocs.io/en/latest/getting-started/installation/

**Comment by JakeYasha** [comment]
Good afternoon!
Thank you very much for the updates and for all the work on Endstone! We have been running our server on Endstone for over a year now, and we are truly happy with the updates released this year.

Thanks to you, our server has fully featured land claims, teleports, coins, friends, trading, mini-games with leaderboards, kits, and even automatic mine regeneration (since players mine resources very quickly — they’re doing great!). We also have bans, SMS auto-translation, ChatGPT and Telegram API integration, clans, PvP, and much more. There is still so much that can be done on the backend side!
We use Redis to store some data and state, and JSON for long-term persistence. We are planning to migrate to Django + PostgreSQL to build an online connection between the database and the store, as well as to achieve compatibility with a Java mirror. Huge thanks to you for all of this! If I manage to get around to it, I’ll publish the code on GitHub as open source.

One thing we really miss — and currently can only solve with workarounds — is backward version compatibility. I managed to make it so players on older versions can connect to the server and fly around, but any interaction causes a crash due to packet incompatibility. If there were a built-in, optionally enabled proxy for handling legacy/current protocol packets, that would be absolutely amazing!
I’ve been able to modify packets “on the fly” manually, but without a proper backend-level abstraction over packet bit layers, it’s very hard to reliably intercept and validate them. If something like this were available out of the box, it would be fantastic 🙂

Thank you so much for your work! You are truly awesome!
